File: typescript-client-sdk/src/sdk/models/shared/notecreate.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// NoteCreate
/** 
 * Create a new Note. You may only associate a note with at most one entity of each type upon creation.
**/
export class NoteCreate extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=accountId" })
  accountId?: string;

  @SpeakeasyMetadata({ data: "json, name=additional" })
  additional?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=contactId" })
  contactId?: string;

  @SpeakeasyMetadata({ data: "json, name=content" })
  content: string;

  @SpeakeasyMetadata({ data: "json, name=dealId" })
  dealId?: string;

  @SpeakeasyMetadata({ data: "json, name=leadId" })
  leadId?: string;

  @SpeakeasyMetadata({ data: "json, name=userId" })
  userId?: string;
}

File: go-client-sdk/pkg/models/shared/notecreate.go
package shared

// NoteCreate
// Create a new Note. You may only associate a note with at most one entity of each type upon creation.
type NoteCreate struct {
	AccountID  *string                `json:"accountId,omitempty"`
	Additional map[string]interface{} `json:"additional,omitempty"`
	ContactID  *string                `json:"contactId,omitempty"`
	Content    string                 `json:"content"`
	DealID     *string                `json:"dealId,omitempty"`
	LeadID     *string                `json:"leadId,omitempty"`
	UserID     *string                `json:"userId,omitempty"`
}
